R/a-helper.R

```diff
@@ -10,33 +10,38 @@ has_input_value <- function(input, input_name) {
 
 
 valid_filters <- function(input, exclude_taxon_rank = TRUE){
-  valid_vars <- c(names(austraits))
   
-  if(exclude_taxon_rank){
-    valid_vars <- valid_vars[valid_vars != "taxon_rank"]
-    stringr::str_subset(names(input), paste(valid_vars, collapse = "|"))
-  } else
-  stringr::str_subset(names(input), paste(valid_vars, collapse = "|"))
+  # Get the names of the input variables
+  v <- names(input)
+  # Limit to values in the database
+  v <- v[v %in% names(austraits)]
+  # remove null values
+  v <- v[purrr::map_lgl(v, ~ !is.null(input[[.x]]))]
```

Comment:
Whoa nice! Have you seen purrr::discard(~is.null)? Might be more readable but this also makes sense - just took a bit of deciphering

Reply:
Nice. Happy to swap to that if you prefer